desc/plotting.py

```diff
@@ -3843,3 +3844,106 @@ def plot_logo(save_path=None, **kwargs):
         fig.savefig(save_path, facecolor=fig.get_facecolor(), edgecolor="none")
 
     return fig, ax
+
+
+def plot_gammac(eq, rhos=None, alphas=None, num_pitch=None):
+    """Plot the energetic proxy metric γ_c fo a single flux surface.
+
+    Parameters
+    ----------
+    eq : object
+        Equilibrium object containing magnetic field information
+    rhos : array_like or float, optional
+        Flux surface radii. If float, plots single surface.
+        Default: np.linspace(0.1, 1, 10)
+    alphas : array_like, optional
+        Fieldline label values (toroidal angle).
+        Default: np.linspace(0, 2π, 32, endpoint=True)
+    num_pitch : int, optional
+        Number of pitch angle values for bounce integral calculation.
+        Default: 16
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The figure object containing the plot
+    ax : matplotlib.axes.Axes
+        The axes object for further customization
+
+    """
+    if rhos is None:
+        rhos = np.array([0.5])
+    elif isinstance(rhos, float):
+        rhos = np.array([rhos])
+
+    print("Calculating γ_c(α, λ)..")
+    if alphas is None:
+        alphas = np.linspace(0, 2 * np.pi, 32, endpoint=True)
+
+    if num_pitch is None:
+        num_pitch = 16
+
+    # TODO(#1352)
+    X, Y = 32, 64
+
+    from desc.integrals.bounce_integral import Bounce2D
+
+    # Compute bounce integral
+    theta = Bounce2D.compute_theta(eq, X, Y, rhos)
+    grid = LinearGrid(rho=rhos, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, sym=False)
+    data0 = eq.compute(
+        "gamma_c",
+        grid=grid,
+        theta=theta,
+        Y_B=64,
+        num_transit=1,
+        num_quad=24,
+        num_pitch=num_pitch,
+        alpha=alphas,
+    )
+    data_full = grid.compress(data0["gamma_c"])
+    print("Plotting ...")
+
+    # Extract pitch angle range
+    minB = data0["min_tz |B|"][0]
+    maxB = data0["max_tz |B|"][0]
+    inv_pitch = np.linspace(minB, maxB, num_pitch)
+
+    # Create figure and prepare colormap
+    fig = plt.figure(figsize=(6, 5))
+    cmap = plt.get_cmap("plasma").copy()
+    cmap.set_under("white")  # Make values below vmin display as white
+
+    # Plot γ_c as function of α and 1/λ
+    extent = [inv_pitch.min(), inv_pitch.max(), alphas.min(), alphas.max()]
+    plt.imshow(
+        data_full[0],
+        origin="lower",  # α increases upward
+        extent=extent,
+        aspect="auto",  # allows aspect ratio to adjust
+        cmap=cmap,
+        interpolation="nearest",  # no smoothing
```

Comment:
why no smoothing? Isn't gamma_c a smooth(ish) function of alpha/lambda? Why not a contour plot?

Reply:
Okay, using contourf.